src/System.IO/tests/Performance/Perf.StreamWriter.cs

```diff
@@ -0,0 +1,116 @@
+            MemoryStream memory = new MemoryStream(new byte[1024 * 1024]);
+            StreamWriter writer = new StreamWriter(memory);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                memory.Position = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        writer.Write("Performance test iteration {0}", i);
+                    }
+                }
+                writer.Flush();
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 20000)]
+        public void Write_Format_OneNoBoxArg()
+        {
+            // Don't want MemoryStream to be expandable to avoid measuring resize
+            MemoryStream memory = new MemoryStream(new byte[1024 * 1024]);
+            StreamWriter writer = new StreamWriter(memory);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                memory.Position = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        writer.Write("Performance test: {0}", nameof(Write_Format_OneNoBoxArg));
+                    }
+                }
+                writer.Flush();
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 20000)]
+        public void WriteLine_Format_OneArg()
+        {
+            // Don't want MemoryStream to be expandable to avoid measuring resize
+            MemoryStream memory = new MemoryStream(new byte[1024 * 1024]);
+            StreamWriter writer = new StreamWriter(memory);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                memory.Position = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        writer.WriteLine("Performance test iteration {0}", i);
+                    }
+                }
+                writer.Flush();
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 20000)]
+        public void Write_Format_Params()
+        {
+            // Don't want MemoryStream to be expandable to avoid measuring resize
+            MemoryStream memory = new MemoryStream(new byte[1024 * 1024]);
+            StreamWriter writer = new StreamWriter(memory);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                memory.Position = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        writer.Write("Performance test iteration {0}", i, i, i, i, i);
```

Comment:
I wanted to get numbers that reflect the overhead of "just" the params, not the actual generation. I'll comment on it.

Reply:
It makes sense for the specific change that you were doing, but, perhaps we should have both: one to measure "just" the params, and one for the typical usage: just to avoid the risk of someone unintentionally optimizing for the "atypical" case.